install.sh

```diff
@@ -33,18 +33,18 @@ fi
 echo "Gateway configuration:"
 
 # Try to get gateway ID from MAC address
-# First try eth0, if that does not exist, try wlan0 (for RPi Zero)
-GATEWAY_EUI_NIC="eth0"
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
-    GATEWAY_EUI_NIC="wlan0"
-fi
 
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
+# Get first non-loopback network device that is currently connected
+NICS=$(ip link 2>&1 | grep "state UP" | grep -v LOOPBACK | sed -E 's/^[0-9]+: ([0-9a-z]+): .*/\1/')
+if [[ -z $NICS ]]; then
     echo "ERROR: No network interface found. Cannot set gateway ID."
     exit 1
 fi
+GATEWAY_EUI_NIC=$(echo $NICS | cut -d " " -f1)
```

Comment:
cutting one could also be done as `| head -1` as part of the bigger one-liner up there ☝️